### PR TITLE
aspects: support "number" type in aspect schemas

### PR DESCRIPTION
**Pull request by MiguelPires**
Adds support for the "number" type in aspect schemas, with constraints to restrict its max, min and allowed values. I used generics to implement a generic validation of the number and int types. I'm not sure if we've used generics before since the codebase predates them but AFAIK we've also never explicitly decided against them. Since this package is new and self-contained, I didn't see a reason not to here.

### CODE REVIEW

**Review by pedronis — APPROVED** [review]
lgtm

File: aspects/schema.go [hunk 1]
```diff
@@ -558,3 +538,93 @@ func (v *anySchema) parseConstraints(constraints map[string]json.RawMessage) err
...
 	return nil
 }
+
+type numberSchema struct {
+	min     *float64
+	max     *float64
+	choices []float64
+}
+
+// Validate that raw is a valid integer and meets the schema's constraints.
```
Comment (ernestl) [inline aspects/schema.go:548]: valid integer -> valid number

**Review by ernestl — APPROVED** [review]
Thanks, looks good, one small comment

**Comment by ernestl** [comment]
The spread failures does not relate to the changes. Force merging.

**Pull request merged**